sympy/polys/matrices/domainmatrix.py

```diff
@@ -166,15 +299,62 @@ def pow(A, n):
             return sqrtAn * sqrtAn
 
     def rref(self):
+        """Return reduced row-echelon form of a Domain matrix and indices of pivot vars.
+        Ensure that the Domain of the input Matrix is a field.
```

Comment:
Rather than "Ensure that..." there could be a "raises" section that explains what error is raised and why.

Reply:
Could you elaborate a bit on the "raises" section and how to go about it.

Reply:
E.g.:
https://github.com/sympy/sympy/blob/6783a0962206feab1cb361eb02ba73e01d3669dd/sympy/calculus/util.py#L58-L63

Reply:
thanks. I actually hadn't seen raises in any docstring so far. That's why I asked